pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -48,6 +48,7 @@ const (
 	PodFinalizer                 = ManagedLabelKey
 	GroupNameLabel               = "kueue.x-k8s.io/pod-group-name"
 	GroupTotalCountAnnotation    = "kueue.x-k8s.io/pod-group-total-count"
+	GroupIndexLabel              = "kueue.x-k8s.io/pod-group-index"
```

Comment:
no strong view here, but imo `pod-group-index` and the var names `PodGroupIndex` are misleadingly suggesting this is index of the pod group. 

I'm thinking about alternatives such as `pod-group-pod-index` or `kueue.x-k8s.io/pod-index`. Any preference here?

Reply:
Fixed